Test/HigherOrderIntrinsicSpecification/ReadPreconditionBypass2.dfy

```diff
@@ -0,0 +1,32 @@
+// RUN: %exits-with 4 %dafny /compile:0 "%s" > "%t"
+// RUN: %diff "%s.expect" "%t"
+
+class Ref {
+  var inner: Ref
+  constructor()
+}
+
+function myf(o: Ref): ()
+  requires false
+  reads o.inner
+{
+  ()
+}
+
+method Main()
+  ensures false
+{
+  var outer := new Ref();
+
+  var hof := myf.reads;
+
+  var inner1 := new Ref();
+  outer.inner := inner1;
+  var reads1 := hof(outer); // Error: function precondition could not be proved
```

Comment:
This comes from replacing`f.reads.requires(t) == true` with` f.reads.requires(t) == f.requires(t)` correct?